amazons3/amazons3_endpoint.bal

```diff
@@ -87,26 +86,33 @@ public remote function Client.getBucketList() returns Bucket[]|error {
 
     request.setHeader(HOST, self.amazonHost);
     request.setHeader(X_AMZ_CONTENT_SHA256, UNSIGNED_PAYLOAD);
-    generateSignature(request, self.accessKeyId, self.secretAccessKey, self.region, GET, requestURI, UNSIGNED_PAYLOAD);
-
-    var httpResponse = self.amazonS3Client->get("/", message = request);
-    if (httpResponse is http:Response) {
-        int statusCode = httpResponse.statusCode;
-        var amazonResponse = httpResponse.getXmlPayload();
-        if (amazonResponse is xml) {
-            if (statusCode == 200) {
-                return getBucketsList(amazonResponse);
+    var signature = generateSignature(request, self.accessKeyId, self.secretAccessKey, self.region, GET, requestURI,
+        UNSIGNED_PAYLOAD);
+
+    if (signature is error) {
+        error err = error(AMAZONS3_ERROR_CODE, { ^"error": signature.detail(),
+            message: "Error occurred while generating the amazon signature header" });
+        return err;
+    } else {
+        var httpResponse = self.amazonS3Client->get("/", message = request);
+        if (httpResponse is http:Response) {
+            int statusCode = httpResponse.statusCode;
+            var amazonResponse = httpResponse.getXmlPayload();
+            if (amazonResponse is xml) {
+                if (statusCode == 200) {
+                    return getBucketsList(amazonResponse);
+                } else {
```

Comment:
We can remove the else bloc and directly say `return setResponseError(statusCode, amazonResponse);` right? Please check other places too.